rfc-153-revert-rfc-126.md

```diff
@@ -0,0 +1,236 @@
+- faraday_middleware (0:03:30)
+```
+</details>
+
+A total of 9,137 Dependabot PRs were opened between 11th June 2020 and 18th July 2022:
+
+- 6,127 (~67%) were from our own internal libraries
+- 855 (~9%) were core requirements for our apps
+- 2,079 (~23%) were other libraries
+
+<details>
+    <summary>Dependabot PRs between June 2020 and July 2022</summary>
+
+```
+Internal libraries:
+Mean time to merge: 3 days, 12:56:00
+Max time to merge: 120 days, 9:17:22
+
+Top 5 slowest libraries to merge:
+- govuk_test (31 days, 11:12:36)
+- plek (24 days, 8:55:07)
+- gds_zendesk (14 days, 2:35:41)
+- stylelint-config-gds (7 days, 3:49:34)
+- govuk_app_config (4 days, 6:15:11)
+
+Top 5 fastest libraries to merge:
+- govuk_publishing_components (1:25:30)
+- gds-api-adapters (0:44:26)
+- govuk_ab_testing (0:40:23)
+- miller-columns-element (0:20:59)
+- govuk_document_types (0:19:04)
+
+Core libraries:
+Mean time to merge: 5 days, 19:02:05
+Max time to merge: 130 days, 9:06:13
+
+Merge times:
+- rspec-rails (14 days, 23:11:42)
+- rails (4:38:04)
+- jasmine (3:33:27)
+- factory_bot_rails (0:59:47)
+
+Other libraries:
+Mean time to merge: 8 days, 4:08:29
+Max time to merge: 419 days, 18:37:27
+
+Top 5 slowest libraries to merge:
+- middleman-gh-pages (419 days, 18:37:27)
+- raven (192 days, 21:14:44)
+- uk-postcode-utils (187 days, 4:58:09)
+- shapely (159 days, 5:03:52)
+- github.com/getsentry/sentry-go (107 days, 2:59:59)
+
+Top 5 fastest libraries to merge:
+- rake (0:03:27)
+- aws-sdk-s3 (0:03:23)
+- lodash (0:02:34)
+- commonmarker (0:02:17)
+- tar (0:01:51)
+```
+</details>
+
+We can conclude that RFC-126 succeeded in reducing the number of Dependabot PRs that did not count as internal or framework dependencies.
+
+## Proposal
+
+Remove all `allow` lists from the `dependabot.yml` file on every GOV.UK repository. This will enable Dependabot PRs [on all dependencies in the lock file][dependabot-docs] of the corresponding [package ecosystem][dependabot-package-ecosystem], including security related PRs.
+
+This is a reversal of RFC-126 as it means every dependency is given equal weighting, rather than us prioritising framework and internal libraries; a key part of the RFC. However, aspects of RFC-126 would remain, namely the Dependabot configuration files that have now been committed to each repository. Without these files, we would [lose all non-security-related version updates][dependabot-docs-delete].
+
+Whilst this proposal will lead to a moderate increase in the number of PRs raised, the increase in continuously deployed apps and the new service teams model should lessen the resulting burden compared with how things were prior to RFC-126. Crucially, we would now receive all security updates as soon as they are available.
+
+Platform Reliability plans to investigate other ways of reducing the effort associated with dependency upgrades, e.g. automatically merging certain PRs, but those ideas for future optimisations are out of scope for this RFC.
```

Comment:
I think it's more within the remit of teams to affect the amount of work to do here now as we have a better ownership model. Teams are now more able to choose to do things like consolidating apps, migrating things to SAAS or auditing/removing unnecessary dependencies.

Reply:
I think the onus is still on Platform Reliability to see how it can help free up teams from worrying about this sort of thing (it's one of our key objectives this quarter!) but you're right that teams have agency to work towards retiring repositories, removing dependencies, or even telling Dependabot to ignore major versions (which would not block security PRs), if they so wish. They can do so via 'ignore' lists (as hinted at in the RFC) or via GitHub comments. 

I can add something to that effect here if you think that would be good 👍

Reply:
I've added on a commit 👍